views/js/qtiCreator/widgets/interactions/graphicGapMatchInteraction/Widget.js

```diff
@@ -115,6 +116,38 @@ define([
             _.forEach(interaction.gapImgs, function(gapImg){
                 $gapList.append(gapImg.render());
             });
+        },
+
+        /**
+         * Display warning message in case any matchMax is set to 0 (infinite)
+         * @param {String} template
+         */
+        infinityMatchMax: function (template) {
+            const interaction = this.element;
+            const response = interaction.getResponseDeclaration();
+
+            const isChoicesInfinityMaxScore = _.some(interaction.getChoices(), function (choice) {
+                return choice.attr('matchMax') === 0;
+            })
+            const isGapImgsInfinityMaxScore = _.some(interaction.getGapImgs(), function (gap) {
+                return gap.attr('matchMax') === 0;
+            })
+            const mappingDisabled = _.isEmpty(response.mapEntries);
+
+            let isInfinityMatchMax = false;
+            switch (template) {
+                case 'hotspot':
+                    isInfinityMatchMax = isChoicesInfinityMaxScore;
+                    break;
+                case 'gapImg':
+                    isInfinityMatchMax = isGapImgsInfinityMaxScore;
+                    break;
+                case 'response':
+                    isInfinityMatchMax = (isChoicesInfinityMaxScore || isGapImgsInfinityMaxScore) && !mappingDisabled;
```

Comment:
```suggestion
                    isInfinityMatchMax = isChoicesInfinityMaxScore && isGapImgsInfinityMaxScore && !mappingDisabled;
```
AND set response processing to “map response“
AND set match max of a hotspot [A] to 0 (infinite)
AND set match max of a gap image [B] to 0 (infinite)
**AND set the score of the response map for the pair hotspot [A] / gap image [B] to a value > 0** - also should be considered this

Reply:
changed!